packages/wallet/test/e2e/SingleAddressWallet/cip30mapping.test.ts

```diff
@@ -0,0 +1,134 @@
+  return await wallet.initializeTx(txProps);
+};
+
+describe('cip30 e2e', () => {
+  let wallet: SingleAddressWallet;
+  let mappedWallet: WalletApi;
+
+  beforeAll(async () => {
+    // CREATE A WALLET
+    const keyAgent = await keyAgentReady;
+    wallet = new SingleAddressWallet(
+      { name: 'Test Wallet' },
+      {
+        assetProvider,
+        keyAgent,
+        stakePoolSearchProvider,
+        timeSettingsProvider,
+        txSubmitProvider,
+        walletProvider
+      }
+    );
+
+    mappedWallet = createCip30WalletApiFromWallet(wallet, {
+      keyAgent,
+      logger: console
+    });
+  });
+
+  afterAll(() => {
+    wallet.shutdown();
+  });
+
+  it('should correctly pull wallet balance correctly from the provider before starting the test', async () => {
+    await firstValueFrom(wallet.balance.total$);
+    expect(wallet.balance.total$.value?.coins).toBeGreaterThanOrEqual(0n);
+  });
+
+  describe('API Methods', () => {
+    test('api.getNetworkId', async () => {
+      const cip30NetworkId = await mappedWallet.getNetworkId();
+      expect(cip30NetworkId).toEqual(Number(process.env.NETWORK_ID));
+    });
+
+    test.skip('api.getUtxos', async () => {
+      expect(async () => await mappedWallet.getUtxos()).not.toThrow();
+    });
+
+    test('api.getBalance', async () => {
+      const balanceCborBytes = Buffer.from(await mappedWallet.getBalance(), 'hex');
+      expect(() => CSL.Value.from_bytes(balanceCborBytes)).not.toThrow();
+    });
+
+    test.skip('api.getUsedAddresses', async () => {
+      const cipUsedAddressess = await mappedWallet.getUsedAddresses();
+      const [{ address: walletUsedAddresses }] = await firstValueFrom(wallet.addresses$);
+
+      const encodedWallet = CSL.Address.from_bech32(cipUsedAddressess[0]);
+
+      expect(encodedWallet).toEqual(walletUsedAddresses);
+    });
+
+    test('api.getUnusedAddresses', async () => {
+      const cipUsedAddressess = await mappedWallet.getUnusedAddresses();
+      expect(cipUsedAddressess).toEqual([]);
+    });
+
+    test('api.getChangeAddress', async () => {
+      const cipChangeAddress = await mappedWallet.getChangeAddress();
+      const [{ address }] = await firstValueFrom(wallet.addresses$);
+      const parsedAddress = parseCslAddress(address as unknown as string);
+      if (!parsedAddress) {
+        throw new Error('No wallet address');
+      }
+
+      expect(cipChangeAddress).toEqual(Buffer.from(parsedAddress.to_bytes()).toString('hex'));
+    });
+
+    test('api.getRewardAddresses', async () => {
+      const cipRewardAddresses = await mappedWallet.getRewardAddresses();
+      const walletRewardAddresses = await wallet.addresses$.value?.filter((s) => s.rewardAccount);
+
+      expect(cipRewardAddresses).toEqual(walletRewardAddresses);
+    });
+
+    test('api.signTx', async () => {
+      const txInternals = await createTxInternals(wallet);
+      expect(await mappedWallet.signTx(cbor.encode(txInternals).toString('hex'))).not.toThrow();
```

Comment:
I don't think this is correct: `Wallet.initializeTx` returns a non-spec-compatible js object. Types defined by cip30 spec should be all supported by CSL and (I think) you can get cbor with (one of) `to_bytes` or `as_bytes`